modules/common/service/service.go

```diff
@@ -67,42 +108,143 @@ func (s *Service) GetServiceHostname() string {
 	return s.serviceHostname
 }
 
-// GetServiceHostnamePort - returns the service hostname with port
-func (s *Service) GetServiceHostnamePort() string {
-	return fmt.Sprintf("%s:%d", s.GetServiceHostname(), GetServicesPortDetails(s.service, s.service.Name).Port)
+// GetServiceHostnamePort - returns the service hostname with port if service port
+// is not nil, otherwise returns GetServiceHostname()
+func (s *Service) GetServiceHostnamePort() (string, string) {
+	servicePort := GetServicesPortDetails(s.service, s.service.Name)
+	if servicePort != nil {
+		return s.GetServiceHostname(),
+			strconv.FormatInt(int64(GetServicesPortDetails(s.service, s.service.Name).Port), 10)
```

Comment:
Since we acquired `servicePort` on L112, couldn't we use it here instead of calling `GetServicesPortDetails(s.service, s.service.Name)` again?

Reply:
right, will change that

Reply:
done